### DISCUSSION TITLE
Converting vel_x and vel_u to vel

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#23271] [Q&A General] [answered]
Hello,

In Navier-Stokes module, because of the way that the kernels are developed, I need to solve for vel_x (velocity in x direction) and vel_y (velocity in y direction) separately (not defining the velocity variable as a vector).

But in a part of my input file, I need to use a kernel which requires me to have velocity as a vector. My question is that how can I define a new variable and convert vel_x and vel_y to a 2D vector called vel?

Thanks,
Ali

**Comment by GiudGiud** [comment, marked as answer]
Hello

I m afraid you can't do that. While you could form a vector auxiliary variable using the component nonlinear variables, you cant use a kernel on that, kernels require nonlinear variables.

The solution here is to re-write the kernel to accept component variables

Guillaume
